### PR TITLE
feat: support automatically translated messages

### LINKED ISSUE #1255: Add support for automatic translations
Add support for automatic translations of messages, currently the SDK doesn't support it.

This will affect MessageText and ChannelPreview

### PR DESCRIPTION
**Pull request by madsroskar**
Our backend supports automatically translated messages - this PR adds support for this in our SDK.

This fixes #1255 

## 🛠 Implementation details

Similarly to the React SDK, this implementation uses the current user language as set in the client (`client.user?.language?`), and will display messages translated to this language if one exists in `message.i18n?[${userLanguage}_text]`.

Where the text is rendered, `message.text` is still used - I figured it was neater to create a new message object with the correct translation set as the text and pass that to the components rendering the text, rather than explicitly looking up the translation in the places it's rendered.

The affected places text is rendered:

- Messages (MessageTextContainer)
- Channel Preview
- Message Overlay

## 🎨 UI Changes

No UI changes are relevant for this

## 🧪 Testing

1. Enable automated translations for the (Stream) app
2. Log in with two different users with different languages set (set `language` in the user object when connecting the user)
3. Send messages from each of the users to eachother in the language they have set 
4. See that the messages are being translated for the recipient

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/src/hooks/useTranslatedMessage.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+type TranslationKey = `${TranslationLanguages}_text`;
+
+export const useTranslatedMessage = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+>(
+  message: MessageResponse<StreamChatGenerics> | FormatMessageResponse<StreamChatGenerics>,
+) => {
+  const { userLanguage } = useTranslationContext();
+  // console.log({ message, userLanguage });
```
Comment (madsroskar) [inline package/src/hooks/useTranslatedMessage.ts:14]:
```suggestion
```

**Comment by vanGalilea** [comment]
We forgot to translate the mesage bubble in the overlay 😳

**Review by vanGalilea — COMMENTED** [review]
We forgot to translate the mesage bubble in the overlay 😳

**Comment by madsroskar** [comment]
Not forgotten, just not acting as expected with the changes to it that are made. Looking into it.

File: package/src/mock-builders/api/getOrCreateChannel.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import { mockedApiResponse } from './utils';
 
+export type GetOrCreateChannelApiParams = {
+  channel?: Record<string, any>;
```
[the remark this answers is not in the document]
Comment (madsroskar) [reply package/src/mock-builders/api/getOrCreateChannel.ts:5]: The generators here are helpers intended to be generic and should just massage data for tests, and I don't think the effort of properly typing it more specifically has any gain or is worth it. 

I might do a pass at improving these in the future, but as this isn't production code I didn't want to spend the time to do much about it on this pass

File: package/src/components/ChannelPreview/hooks/useLatestMessagePreview.ts [hunk 3]
```diff
@@ -207,7 +207,10 @@ const getLatestMessagePreview = <
       status: MessageReadStatus.NOT_SENT_BY_CURRENT_USER,
     };
   }
-  const message = lastMessage || messages.length ? messages[messages.length - 1] : undefined;
+
+  const channelStateLastMessage = messages.length ? messages[messages.length - 1] : undefined;
+
+  const message = lastMessage !== undefined ? lastMessage : channelStateLastMessage;
```
Comment (santhoshvai) [inline package/src/components/ChannelPreview/hooks/useLatestMessagePreview.ts:213]:
```suggestion
  const message = lastMessage ?? channelStateLastMessage;
```

obviously an optional suggestion

Comment (santhoshvai) [reply package/src/components/ChannelPreview/hooks/useLatestMessagePreview.ts:213]: nah, lets leave it there ☠️

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**